crypto/encode_decode/encoder_pkey.c

```diff
@@ -327,6 +327,14 @@ OSSL_ENCODER_CTX *OSSL_ENCODER_CTX_new_for_pkey(const EVP_PKEY *pkey,
         && OSSL_ENCODER_CTX_set_selection(ctx, selection)
         && ossl_encoder_ctx_setup_for_pkey(ctx, pkey, selection, propquery)
         && OSSL_ENCODER_CTX_add_extra(ctx, libctx, propquery)) {
+        OSSL_PARAM params[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
+        int save_parameters = pkey->save_parameters;
+
+        params[0] = OSSL_PARAM_construct_int(OSSL_ENCODER_PARAM_SAVE_PARAMETERS,
+                                             &save_parameters);
+        /* ignoring error as this is only auxiliary parameter */
+        (void)OSSL_ENCODER_CTX_set_params(ctx, params);
```

Comment:
What is the rationale for ignoring errors from this?
Unknown parameters are ignored and a success returned.  An error indicates that something has gone tangibly wrong with the request.

Reply:
In theory an encoder could "know" the parameter and decide to reject its value. Of course our encoders do not do that but third party could do. As this is really an auxiliary stuff I do not think we want to fail the encoding in such case.